addons/mail/static/src/models/thread/thread.js

```diff
@@ -168,6 +168,20 @@ function factory(dependencies) {
                     data2.isServerPinned = true;
                 }
             }
+            if ('last_message' in data && data.last_message) {
+                if (!data2.messagesAsServerChannel) {
+                    data2.messagesAsServerChannel = [['unlink-all']];
```

Comment:
Note that  `[['unlink-all']]` is absolutely not correct. Receiving a last message should not discard potentially existing other messages.